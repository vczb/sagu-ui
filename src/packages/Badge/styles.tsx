import styled from 'styled-components'
import { BadgeProps } from '.'

export const Badge = styled.div<Pick<BadgeProps, 'size' | 'shape' | 'color'>>`
  display: inline-block;
  border-radius: ${({ shape }) =>
    shape === 'circle' ? '50%' : shape === 'rectangle' ? '0px' : '0px'};
  font-size: ${({ size }) =>
    size === 'xsmall'
      ? '0.6em'
      : size === 'small'
      ? '0.7em'
      : size === 'medium'
      ? '0.8em'
      : '1em'};
  padding: ${({ size, shape }) =>
    shape === 'circle' && (size === 'xsmall' || size === 'small')
      ? '0.4em'
      : '0.2em 0.5em'};
  background-color: ${({ color, theme }) =>
    color || theme.colors.primary.medium};
  color: ${({ theme }) => theme.colors.base.white};
`
