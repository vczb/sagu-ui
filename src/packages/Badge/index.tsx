import React from 'react'
import { HTMLAttributes } from 'react'
import * as S from './styles'

export type BadgeProps = {
  content: string | number
  size?: 'xsmall' | 'small' | 'medium' | 'large'
  shape?: 'circle' | 'square' | 'rectangle'
  color?: string
} & HTMLAttributes<HTMLDivElement>

const Badge: React.FC<BadgeProps> = ({
  content,
  size = 'medium',
  shape = 'square',
  color,
  ...props
}) => (
  <S.Badge size={size} shape={shape} color={color} {...props}>
    {content}
  </S.Badge>
)

export default Badge
