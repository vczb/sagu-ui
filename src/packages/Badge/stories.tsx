import React from 'react'
import { StoryFn, Meta } from '@storybook/react'
import Badge, { BadgeProps } from '.'

export default {
  title: 'Badge',
  component: Badge,
  args: {
    content: 'New',
    size: 'medium',
    shape: 'square'
  }
} as Meta

export const Default: StoryFn<BadgeProps> = (args) => <Badge {...args} />
