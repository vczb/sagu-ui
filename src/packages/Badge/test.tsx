import React from 'react'
import { render } from '../../utils/testUtils'

import Badge from '.'

describe('<Badge />', () => {
  it('should render Badge with default props', () => {
    const { container } = render(<Badge content="5" />)
    expect(container).toMatchSnapshot()
  })

  it('should render Badge with custom size, shape, and color', () => {
    const { container } = render(
      <Badge content="New" size="large" shape="circle" color="red" />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render a medium small circle badge', () => {
    const { container } = render(
      <Badge content="New" size="small" shape="circle" color="red" />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render Badge by handling custom props correctly', () => {
    const { getByTestId } = render(
      <Badge data-testid="custom-badge" title="Custom Title" content="Custom" />
    )
    const badge = getByTestId('custom-badge')
    expect(badge).toHaveAttribute('title', 'Custom Title')
  })
})
