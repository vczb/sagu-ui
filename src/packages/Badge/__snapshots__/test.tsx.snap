// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render Badge with custom size, shape, and color 1`] = `
.c0 {
  display: inline-block;
  border-radius: 50%;
  font-size: 1.8rem;
  padding: 0.8rem 1.6rem;
  background-color: red;
  color: #F8F7FB;
}

<div>
  <div
    class="c0"
    color="red"
    shape="circle"
  >
    New
  </div>
</div>
`;

exports[`<Badge /> should render Badge with default props 1`] = `
.c0 {
  display: inline-block;
  border-radius: 0;
  font-size: 1.6rem;
  padding: 0.8rem 1.6rem;
  background-color: #022B54;
  color: #F8F7FB;
}

<div>
  <div
    class="c0"
    shape="square"
  >
    5
  </div>
</div>
`;

exports[`<Badge /> should render a medium small circle badge 1`] = `
.c0 {
  display: inline-block;
  border-radius: 50%;
  font-size: 1.4rem;
  padding: 0.4rem;
  background-color: red;
  color: #F8F7FB;
}

<div>
  <div
    class="c0"
    color="red"
    shape="circle"
  >
    New
  </div>
</div>
`;
